00_Base/src/model/ServerCredentialsRole.ts

```diff
@@ -77,4 +78,22 @@ export class ServerCredentialsRole extends Model implements ICredentialsRole {
     serverCredentialsRole.business_details = businessDetails;
     return serverCredentialsRole;
   }
+
+  static toCredentialsRoleDTO(
+    serverCredentialsRole: ServerCredentialsRole,
+  ): CredentialsRoleDTO {
+    const credentialsRoleDTO = new CredentialsRoleDTO();
+    credentialsRoleDTO.role =
+      serverCredentialsRole[ServerCredentialsRoleProps.role];
```

Comment:
I would say that if we don't need to use the "Props" then maybe we shouldn't for the sake of readability.

Reply:
the issue is that we need props because the lazy loading with `Model.$get('fieldName')` takes in that field name string and if I keep it as a string, then in the future if the model is to change, it will introduce bugs, so there needs to be some sort of constants for sequelize models =/ but using Props in general helps in other situations as well as it helps ensure that the property name is centralized...

Reply:
If we use the way of `model.fieldName` here, it is easy to read. And when the fieldName is changed, there will be a build error so that we still have a chance to notice it. From this perspective, it is safer than `Model.$get('fieldName')`. But using props will save more effort, i.e, we just need to change one place when update fieldName. Personally, I would prefer to using props.

Reply:
Using `mode.fieldName` won't lazy load and query the DB to get the association thought right? I think you do need `.$get(..)` at least in some places where it makes sense... Also, wouldn't `.$get(..)` only error out during runtime? I was thinking that this could introduce a hard to find bug that will only happen when this code runs..